content/en-us/luau/type-checking.md

```diff
@@ -0,0 +1,326 @@
+
+```lua
+part.Touched:Connect(function(otherPart: BasePart)
+	local brickColor: Enum.BrickColor = otherPart.BrickColor
+end)
+```
+
+An optional type can be defined using `?` at the end of the annotation.
+
+```lua
+local foo: string? = nil
+```
+
+## Function Typing
+
+Assume the following function
+```lua
+local function add(x, y)
+	return x + y
+end
+```
+
+This is a fairly simple function where the behavior is well defined. Take the value of `x` and add it to `y`. However, nothing is stopping someone from using a `string` type in the `add` function, which causes an error.
+
+This is because Luau doesn't know that we want to use numbers in this function, lets rework this function such that it does constrain `x` and `y` to be numbers.
+
+```lua
+local function add(x: number, y: number)
+	return x + y
+end
+```
+
+By adding type annotations to the parameters of the function, Luau now knows that the function takes two numbers, and throws a warning if you try to put anything into the function that isn't a number
+
+```lua
+add(5, 10) -- ok
+add(5, "foo") -- TypeError: Type `string` could not be converted into `number`.
+```
+
+To define a return type, put a `:` operator at the end of the function definition
+
+```lua
+local function add(x: number, y: number): number
+```
+
+To return multiple types, place all return types in brackets
+
+```lua
+local function FindSource(script: BaseScript, pattern: string): (string, number)
+```
+
+### Defining a Functional Type
+
+A functional type can be defined by using the syntax `(in) -> out`, for example, the type signatures of the functions above is
+
+```lua
+type add = (x: number, y: number) -> number
+type FindSource = (script: BaseScript, pattern: string) -> (string, number)
+```
+
+## Table Types
+
+Luau does not have a simple `table` type, instead, table types are defined using `{}` syntax. At the most basic level, a table can be defined with the syntax `{type}`
+
+```lua
+local numbers: {number} = {1, 2, 3, 4, 5}
+```
+
+Index types can be defined using `{[indexType] = valueType}`
+
+```lua
+local messagesFromPeople: {[string]: any} = {
+	Dave = "Hello",
+	Cat = "meow"
+}
+```
+
+Tables can also have explicit string indicies defined in a type
+
+```lua
+type Car = {
+	Speed: number,
+	Drive: (Car) -> ()
+}
+```
+
+## Variadics
+
+Lets assume a function that calculates the sum of a variadic ammount of numbers
+
+```lua
+local function addLotsOfNumbers(...)
+	local sum = 0
+
+	for _, v in {...} does
+		sum += v
+	end
+
+	return sum
+end
+```
+
+As expected, this can take any value and wont complain if you give it an invalid type such as a `string`.
+
+```lua
+print(addLotsOfNumbers(1, 2, 3, 4, 5)) -- 15
+print(addLotsOfNumbers(1, 2, "3", 4, 5)) -- attempt to add string to number
+```
+
+Instead, we can assign a type to the `...` just like how you'd assign any other type

```

Comment:
```suggestion
Instead, you can assign a type to `...` just like how you'd assign any other type.
```

Use of the word "we" refers to Roblox and implies a recommendation. Check style guide.

I did not make a suggestion against "just like how you'd assign any other type" though I do feel it could be chopped off or rephrased, though you can resolve this if you feel it's fine as it is.